docs/howto/add_link.md

```diff
@@ -0,0 +1,336 @@
+[Back to Guides](../README.md)
+
+# How to add links
+
+`ActiveModelSerializers` offers you many ways to add links in your JSON reponse depending on the adapter you are using.
+
+Note that within the following examples, we set the `adapter` option, but that is not needed if the `ActiveModelSerializers.config.adapter` variable is properly set. Also, some examples are given for both Rails and [non-Rails usage](outside_controller_use.md).
+
+Unless stated otherwise, the examples given here will assume that the following classes are defined:
+
+```ruby
+class User < ActiveModelSerializers::Model
+  attributes :id, :name
+end
+
+class UserSerializer < ActiveModel::Serializer
+  attributes :id, :name
+end
+```
+
+## JSON API adapter
+
+The JSON API specification allows the usage of the `links` member for representing links in three cases:
+* [Top level links](http://jsonapi.org/format/#document-top-level)
+* [Resource link](http://jsonapi.org/format/#document-resource-object-links)
+* [Relationship links](http://jsonapi.org/format/#document-resource-object-relationships)
+
+`ActiveModelSerializers` provides tools to handle these links.
+
+### [Top level links](http://jsonapi.org/format/#document-top-level)
+
+In order to provide top-level links, you will need to specify the `links` option to the serialization:
+
+```ruby
+user = User.new(id: 1, name: "John")
+links = { home: "https://www.example.com/" }
+resource = ActiveModelSerializers::SerializableResource.new(user, adapter: :json_api, links: links)
+resource.to_json
+```
+
+This would result in:
+
+```json
+{
+  "links": {
+    "home": "https://www.example.com/"
+  },
+  "data": {
+    "id": "1",
+    "type": "users",
+    "attributes": {
+      "name": "John"
+    }
+  }
+}
+```
+
+When using Rails, you can add top-level links by providing the `links` option to `render` method:
+
+```ruby
+class UserController < ActionController::Base
+  def show
+    user = User.new(id: 1, name: "John")
+    links = { home: "https://www.example.com/" }
+    render json: user, adapter: :json_api, links: links
+  end
+end
+```
+
+#### Pagination
+
+`ActiveModelSerializers` will provide automatic pagination links for collections if you use [Kaminari](https://github.com/amatsuda/kaminari)
+or [WillPaginate](https://github.com/mislav/will_paginate) and the `ActiveModelSerializers.config.jsonapi_pagination_links_enabled` config is set to `true` (defaults to `true`). To disable this feature, you can simply set this config to `false`.
+
+```ruby
+class UserController < ActionController::Base
+  def kaminari
+    users = User.all.to_a # .all.to_a used as example
```

Comment:
And example could just be, assuming we're rendering the third page, one item per page

```ruby
render json: users, adapter: :json_api
```

would render as: